cvmfs/swissknife_check.cc

```diff
@@ -566,15 +598,24 @@ bool CommandCheck::Find(const catalog::Catalog *catalog,
         // are all data chunks in the data store?
         if (check_chunks_ && !entries[i].IsExternalFile()) {
           const shash::Any &chunk_hash = this_chunk.content_hash();
-          const string chunk_path = "data/" + chunk_hash.MakePath();
-          if (!Exists(chunk_path)) {
-            LogCvmfs(kLogCvmfs, kLogStderr, "partial data chunk %s (%s -> "
-                                            "offset: %d | size: %d) missing",
-                     this_chunk.content_hash().ToStringWithSuffix().c_str(),
-                     full_path.c_str(),
-                     this_chunk.offset(),
-                     this_chunk.size());
-            retval = false;
+          // for performance reasons, only perform the check once
+          // and skip if the hash has been checked before
+         bool chunk_needs_check = true;
+          if (!no_duplicates_map_ && duplicates_map_.Contains(chunk_hash)) {
```

Comment:
Here seems to be two issues

- Whitespacing (also following lines)
- Logic: we should insert the chunk if the map does _not_ contain it already.

Reply:
True, thanks!